pkg/agents/team6/communication.go

```diff
@@ -0,0 +1,254 @@
+// The altruist and the collectivist do not request anything like that
+// The selfish and the narcissist request the other agent to take nothing
+func (a *CustomAgent6) regainTrustInNeighbours() {
+
+	if a.trustTeams[a.neighbours.above] < -10 {
+		msg := messages.NewRequestTakeFoodMessage(a.ID(), a.Floor(), a.Floor()-1, 0)
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "RequestLeaveFood", "floor": a.Floor()})
+	}
+
+	if a.trustTeams[a.neighbours.below] < -10 {
+		msg := messages.NewRequestTakeFoodMessage(a.ID(), a.Floor(), a.Floor()+1, 0)
+		a.SendMessage(msg)
+		a.Log("I sent a message", infra.Fields{"message": "RequestLeaveFood", "floor": a.Floor()})
+	}
+}
+
+func (a *CustomAgent6) HandleResponse(msg messages.BoolResponseMessage) {
+	if msg.Response() {
+		a.trustTeams[msg.SenderID()] = 0
+	} else {
+		a.updateTrust(-1, msg.SenderID())
+	}
+}
+
+// Handles RequestLeaveFood messages the agent receives
+// Returns true if the agent accepts the request, false otherwise
+func (a *CustomAgent6) HandleRequestLeaveFood(msg messages.RequestLeaveFoodMessage) {
+	healthInfo := a.HealthInfo()
+	currentHP := a.HP()
+
+	// HP levels based on maximim HP
+	levels := levelsData{
+		strongLevel:  healthInfo.MaxHP * 3 / 5,
+		healthyLevel: healthInfo.MaxHP * 3 / 10,
+		weakLevel:    healthInfo.MaxHP * 1 / 10,
+	}
+
+	var reply bool
+
+	switch a.currBehaviour.string() {
+	case "Altruist":
+		reply = true
+
+	case "Collectivist":
+		if currentHP >= levels.weakLevel {
+			reply = true
+		} else {
+			reply = false
+		}
+
+	case "Selfish":
+		if currentHP >= levels.strongLevel {
+			reply = true
+		} else {
+			reply = false
+		}
+
+	case "Narcissist":
+		reply = false
+		a.updateTrust(-1, msg.SenderID()) // how dare you even ask
+
+	default:
+		reply = true
+	}
+
+	replyMessage := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), reply)
+	a.SendMessage(replyMessage)
+
+	if reply {
+		a.reqLeaveFoodAmount = msg.Request()
+		a.Log("I received a requestLeaveFood message and my response was true")
+	} else {
+		a.reqLeaveFoodAmount = -1
+		a.Log("I received a requestLeaveFood message and my response was false")
+	}
+	// Try to identify our neighbours
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+// Handles RequestTakeFood messages the agent receives
+// Returns false, as our agents never accept to take a precise, fixed amount of food
+func (a *CustomAgent6) HandleRequestTakeFood(msg messages.RequestTakeFoodMessage) {
+	reply := false
+	replyMessage := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), reply)
+	a.SendMessage(replyMessage)
+
+	if reply {
+		a.reqLeaveFoodAmount = msg.Request()
+		a.Log("I received a requestTakeFood message and my response was true")
+	} else {
+		a.reqLeaveFoodAmount = -1
+		a.Log("I received a requestTakeFood message and my response was false")
+	}
+	// Try to identify our neighbours
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
```

Comment:
??

Dead code
Just pass `false` directly into `Reply()`
And you don't need the if statement, since the if block will never execute, only the else

Reply:
False was a placeholder - updating now (`sorry for this :((( `)

Reply:
We updated it. It does not always return false now